### PR TITLE
Popular species & refactoring species_stats

### PR DESCRIPTION
**Pull request by kamaldodiya**
**This PR**
-  Implements new endpoint /api/metadata/popular_species endpoint
- Refactors the field_name for the stats endpoint due to upstream changes in gRPC message

**Please note that** 
- The data for pseudogene_stats are missung as data is missing in gRPC service - Refer EA-1111
-  homology, variants data is missing in gRPC service as well
 
**Before merging PR**

- Production Team
    - The data on meta-prod-replica should be available (Schema Changes) 
    - The packages version for ensembl-metadata-api and Ensembl-metadata-service should be updated for bringing code changes due to schema change
- Web Team
    - genome images should be copied to nginx static service - Done by @azangru 

**Related JIRA**
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2180
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2190
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2191

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,11 +7,8 @@ license = "Apache-2.0"
 
 [tool.poetry.dependencies]
 python = "^3.8"
-fastapi = "^0.57.0"
-pydantic = "^1.5.1"
-starlette = "^0.13.4"
 loguru = "^0.5.1"
-ensembl-metadata-service = {git = "https://github.com/Ensembl/ensembl-metadata-service.git", rev = "0.0.2b1"}
+ensembl-metadata-service = {git = "https://github.com/Ensembl/ensembl-metadata-service.git", rev = "hotfix/fix_build_pipline"}
```
Comment (kamaldodiya) [inline pyproject.toml:11]: This will be changed before merging PR to whatever version production releases.

File: app/api/models/statistics.py [hunk 2]
```diff
@@ -111,12 +111,15 @@ class GenomeStatistics(BaseModel):
...
-            stats_item["name"]: stats_item["statisticValue"]
-            for stats_item in data["_raw_data"]
-            if stats_item["statisticValue"] != "null"
-        }
-
+        data["_compiled_data"] = {}
+        try:
+            for stats_item in data["_raw_data"]:
+                try:
+                    data["_compiled_data"][stats_item["name"]]=stats_item["statisticValue"]
```
Comment (Mehrnaz-Charkhchi) [inline app/api/models/statistics.py:118]: Can we use .get() to avoid the try: except and to help the code to survive in these cases?

File: app/api/resources/metadata.py [hunk 3]
```diff
@@ -44,5 +45,17 @@ async def get_metadata_statistics(request: Request, genome_uuid: str):
...
     except Exception as e:
-        logger.log("INFO", e)
+        return response_error_handler({"status": 500})
+
+@router.get("/popular_species", name="popular_species")
+async def get_popular_species(request: Request):
+    try:
+        popular_species_dict = MessageToDict(grpc_client.get_popular_species())
+        popular_species = popular_species_dict['organismsGroupCount']
+        for species in popular_species:
```
Comment (Mehrnaz-Charkhchi) [inline app/api/resources/metadata.py:55]: I'm wondering if this part can be done in models while the objects are being created using a Pydantic @validator .

File: app/api/models/statistics.py (hunk 2 above)
Comment (kamaldodiya) [reply app/api/models/statistics.py:118]: That's reasonable thing to do but keeping try..except will give us an opportunity to find out which fields value is missing. and it will survive the missing fields.

File: app/api/resources/metadata.py (hunk 3 above)
Comment (kamaldodiya) [reply app/api/resources/metadata.py:55]: I wish I could but in order to get the base_url we have to modify it here or else we will have to pass the request object or base_url with all the request. Which I think is overkill.

File: pyproject.toml [hunk 4]
```diff
@@ -7,11 +7,8 @@ license = "Apache-2.0"
 
 [tool.poetry.dependencies]
 python = "^3.8"
-fastapi = "^0.57.0"
```
Comment (Mehrnaz-Charkhchi) [inline pyproject.toml:10]: Removing these 3 lines is causing the project build to break.
please revert this change

Comment (Mehrnaz-Charkhchi) [reply pyproject.toml]: @kamaldodiya Did you try to build a new docker image?
20.82 Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.
34.28 
34.28 Because ensembl-web-metadata-api depends on starlette (^0.13.4) which doesn't match any versions, version solving failed.
------
failed to solve: process "/bin/sh -c pip install poetry &&     poetry config virtualenvs.create false &&     poetry install --no-dev" did not complete successfully: exit code: 1

**Review by veidenberg — APPROVED** [review]

File: app/api/resources/metadata.py (hunk 3 above)
Comment (Mehrnaz-Charkhchi) [reply app/api/resources/metadata.py:55]: It would be better to follow the MVC design patter and keep data logic in the models.
passing request.url as a string to the model is very similar to passing  species.image in the updated dictionary.
 The only change is that we move the data logic to the model.

Comment (kamaldodiya) [reply app/api/resources/metadata.py:55]: Model expects image as input. Note that the image would base_url + <species_taxonomy_id>.svg. Moreover image is string so I can not assign speciesTaxonomyId since it is number. I could convert it to str and then assign it and later in field_serialiser it will give the URL. 
So we have 2 options 
1) pass base_url, convert species_taxonomy_id to str and assign it to image , implement steriliser
2) just form the URL based on the species_taxonomy_id and pass it to model

Comment (kamaldodiya) [reply app/api/resources/metadata.py:55]: Alternatively, one more option is 

1. to introduce class variable base_url which will be set once/request.
2. Make image Nullable so while generating the PopularSpecies it won't have image
3. Implement serialised and for the URL based on the class variable base_url and species_taxonomy_id

Comment (Mehrnaz-Charkhchi) [reply app/api/resources/metadata.py:55]: Serialisers are best when we want to modify the output of a model which would cause the issues that you just mentioned.
We can add a root_validator to access all the values before they are assigned to fields, for either PopularSpeciesGroup or PopularSpecies.
 For PopularSpeciesGroup to do similar to what you have done in the view would look like the following:
Model:
`class PopularSpeciesGroup(BaseModel):
    _request_url: str  # A hidden field to store request.url
    popular_species: List[PopularSpecies]

    @root_validator(pre=True)
    def generate_objects(cls, values):
        logger.info(values)
        for species in values["popular_species"]:
            species['image'] = '{}static/genome_images/{}.svg'.format(values["_request_url"],
                                                                      species['speciesTaxonomyId'])
        return values`

and the view would look like this: 
` popular_species_response = PopularSpeciesGroup(_request_url=request.base_url,popular_species=popular_species)`

We can do exactly the same in PopularSpecies and it wouldn't need the for loop anymore because data will be updated automatically before it's assigned to the fields.

File: app/api/models/statistics.py (hunk 2 above)
Comment (Mehrnaz-Charkhchi) [reply app/api/models/statistics.py:118]: I think some of these try excepts are not allowing the code to have standard outputs that can help easily identify the issues specially when we're on k8s. For example: I couldn't even see that GRPC is broken at the moment and everything seem to be fine in the logs except a 500 Error!
I won't get any sot of useful information in my logs unless I remove try: excepts to see where the code is exactly. 
Keep that in mind that in the next update I'm going to change this in web_metadata_api but we can go ahead with this for now.

**Review by Mehrnaz-Charkhchi — APPROVED** [review]
